core/src/main/scala/org/bitcoins/core/wallet/utxo/TxoState.scala

```diff
@@ -14,6 +14,8 @@ object TxoState extends StringFactory[TxoState] {
   /** Means that no funds have been sent to this utxo EVER */
   final case object DoesNotExist extends TxoState
 
+  final case object ImmatureCoinbase extends TxoState
```

Comment:
This seems like  a great idea! I would add a scaladoc to it refering to what it is exactly with a link to some external documentation too